src/chess/bitboard.h

```diff
@@ -190,40 +190,64 @@ class Move {
  public:
   enum class Promotion : std::uint8_t { None, Queen, Rook, Bishop, Knight };
   Move() = default;
-  Move(BoardSquare from, BoardSquare to) : from_(from), to_(to) {}
-  Move(BoardSquare from, BoardSquare to, Promotion promotion)
-      : from_(from), to_(to), promotion_(promotion) {}
-  Move(const std::string& str, bool black = false);
-  Move(const char* str, bool black = false) : Move(std::string(str), black) {}
-
-  BoardSquare from() const { return from_; }
-  BoardSquare to() const { return to_; }
-  Promotion promotion() const { return promotion_; }
-  bool IsCastling() const { return castling_; }
-  void SetCastling() { castling_ = true; }
+  Move(BoardSquare from, BoardSquare to) {
```

Comment:
This constructor and the next one are quite performance critical and `SetTo()` functions do unnecesary job (clearing bits, for example).
I'd suggest to write directly to data_ in constructor:
`: data_(to.as_int() + (from.as_int() << 6)){}`

Reply:
I had suggested exactly the opposite, which is why it's this way now. Is it really that important? There are already data with the current code that shows no performance difference.

Reply:
Done